tests/unit/ad/test_grid_operators.py

```diff
@@ -248,30 +247,31 @@ def test_boundary_grid_projection(mdg: pp.MixedDimensionalGrid, scalar: bool):
     # Compute geometry for the mixed-dimensional grid. This is needed for
     # boundary projection operator.
     mdg.compute_geometry()
-    projection = pp.ad.grid_operators.BoundaryProjection(
-        mdg, mdg.subdomains(), proj_dim
-    )
+    projection = pp.ad.BoundaryProjection(mdg, mdg.subdomains(), proj_dim)
+    # Obtaining sparse matrices from the AD Operators.
+    subdomain_to_boundary = projection.subdomain_to_boundary.parse(mdg)
+    boundary_to_subdomain = projection.boundary_to_subdomain.parse(mdg)
     # Check sizes.
-    assert projection.subdomain_to_boundary().shape == (num_cells, num_faces)
-    assert projection.boundary_to_subdomain().shape == (num_faces, num_cells)
+    assert subdomain_to_boundary.shape == (num_cells, num_faces)
+    assert boundary_to_subdomain.shape == (num_faces, num_cells)
 
     # Check that the projection matrix for the top-dimensional grid is non-zero.
     # The matrix has eight boundary faces.
     ind0 = 0
     ind1 = g_0.num_faces * proj_dim
-    assert np.sum(projection.subdomain_to_boundary()[:, ind0:ind1]) == 8 * proj_dim
+    assert np.sum(subdomain_to_boundary[:, ind0:ind1]) == 8 * proj_dim
     # Check that the projection matrix for the first fracture is non-zero. Since the
     # fracture touches the boundary on two sides, we expect two non-zero rows.
     ind0 = ind1
     ind1 += g_1.num_faces * proj_dim
-    assert np.sum(projection.subdomain_to_boundary()[:, ind0:ind1]) == 2 * proj_dim
+    assert np.sum(subdomain_to_boundary[:, ind0:ind1]) == 2 * proj_dim
     # Check that the projection matrix for the second fracture is non-zero.
     ind0 = ind1
     ind1 += g_2.num_faces * proj_dim
-    assert np.sum(projection.subdomain_to_boundary()[:, ind0:ind1]) == 2 * proj_dim
+    assert np.sum(subdomain_to_boundary[:, ind0:ind1]) == 2 * proj_dim
     # The projection matrix for the intersection should be zero.
     ind0 = ind1
-    assert np.sum(projection.subdomain_to_boundary()[:, ind0:]) == 0
+    assert np.sum(subdomain_to_boundary[:, ind0:]) == 0
```

Comment:
We don't have any tests of boundary_to_subdomain (I know, it is my fault). I think this should be the transpose, so perhaps we can add a test 

assert np.allclose((subdomain_to_boundary - boundary_to_subdomain.T).data, 0)

Or roughly so.

Reply:
Added